SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -1215,10 +1412,23 @@ void CFlowOutput::SetVolumeOutputFields_ScalarLimiter(const CConfig* config) {
     }
   }
 
-  if (config->GetKind_Species_Model() != SPECIES_MODEL::NONE) {
-    if (config->GetKind_SlopeLimit_Species() != LIMITER::NONE) {
-      for (unsigned short iVar = 0; iVar < config->GetnSpecies(); iVar++)
-        AddVolumeOutput("LIMITER_SPECIES_" + std::to_string(iVar), "Limiter_Species_" + std::to_string(iVar), "LIMITER", "Limiter value of the transported species " + std::to_string(iVar));
+  if (config->GetKind_SlopeLimit_Species() != LIMITER::NONE) {
+    switch (config->GetKind_Species_Model()) {
+      case SPECIES_MODEL::SPECIES_TRANSPORT:
+        for (unsigned short iVar = 0; iVar < config->GetnSpecies(); iVar++)
+          AddVolumeOutput("LIMITER_SPECIES_" + std::to_string(iVar), "Limiter_Species_" + std::to_string(iVar), "LIMITER", "Limiter value of the transported species " + std::to_string(iVar));
+      break;
+      case SPECIES_MODEL::FLAMELET:
+        AddVolumeOutput("LIMITER_PROGVAR", "Limiter_Progress_Variable", "SOLUTION", "Limiter of progress variable");
+        AddVolumeOutput("LIMITER_ENTHALPY", "Limiter_Total_Enthalpy", "SOLUTION", "Limiter of total enthalpy");
+        /*--- limiter for auxiliary species transport ---*/
+        for (unsigned short iReactant=0; iReactant<config->GetNUserScalars(); iReactant++) {
+          string species_name = config->GetUserScalarName(iReactant);
+          AddVolumeOutput("LIMITER_" + species_name, "LIMITER_" + species_name, "LIMITER", "Limiter value for the " + species_name + " equation");
+        }
```

Comment:
This should not be added here it may break restarts, add only the solution and add the other fields in another function. Setting the values is fine regardless of order.

Reply:
I don't understand what the problem is here. Are you talking specifically about the limiters for the auxiliary species or also total enthalpy? How is it different from the other AddVolumeOutputs?

Reply:
The issue is that the order in which groups (such as solution) are added matters for restart purposes.
Therefore, we need to write all the solutions, then primitives etc.
I may have put my comment in the wrong place but here you are putting limiters in the solution group, so that's probably not correct.

Reply:
OK, I am putting scalar primitive, limiter, source and lookup together in their own functions.  These functions are now small but will become a little larger with Everts PR for partially premixed combustion later